mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+            function_values = []  # Array of pairwise interactions [f_k_l(not centered), f_k(centered), f_l(centered)]
+            col_k = pair[0]
+            col_l = pair[1]
+
+            y_cdf_k_centered = self.ind_partial_dep_functions[col_k] - np.mean(self.ind_partial_dep_functions[col_k])
+            y_cdf_l_centered = self.ind_partial_dep_functions[col_l] - np.mean(self.ind_partial_dep_functions[col_l])
+
+            for x_k, y_cdf_k in zip(self.feature_values[col_k], y_cdf_k_centered):
+                for x_l, y_cdf_l in zip(self.feature_values[col_l], y_cdf_l_centered):
+                    col_k_position = self.feature_column_position_mapping[col_k]
+                    col_l_position = self.feature_column_position_mapping[col_l]
+                    X_ = self.X.values.copy()
+                    X_[:, col_k_position] = x_k
+                    X_[:, col_l_position] = x_l
+
+                    y_pred = self._get_model_predictions(X_)
+                    y_cdf_k_l = y_pred.mean()
+
+                    function_values.append([y_cdf_k_l, y_cdf_k, y_cdf_l])
+
+            function_values = np.array(function_values)  # Convert to np.array to vectorize operations
+
+            # Cumulative pairwise effect value for a given feature
+            # Need to center f_k_l as f_k, f_l have been already centered
+            func_value = sum(abs((function_values[:, 0] - np.mean(function_values[:, 0]) - function_values[:,
+                                                                                                           1] - function_values[:, 2])))
+
+            store[str(pair)] = func_value / (self.num_values ** 2)
+
+        self.pair_wise_effect = {'raw': store, 'norm': _normalize(store)}
+
+    def fit(self) -> None:
+        """
+        Get linear, non-linear effects estimation.
+
+        :return: None
+        """
+        linear_effect = self._get_linear_effect_estimation()
+        non_linear_effect = self._get_non_linear_effect_estimation()
+
+        self.linear_effect = {'raw': linear_effect, 'norm': _normalize(linear_effect)}
+        self.non_linear_effect = {'raw': non_linear_effect, 'norm': _normalize(non_linear_effect)}
+
+    def plot_effects(self) -> None:
+        """
+        Plot each effect (normalized) on a bar plot (linear, non-linear). Also plots pairwise effects if calculated.
+
+        :return: (plt.figure) plot figure
+        """
+        if self.pair_wise_effect is None:
+            fig, (ax1, ax2) = plt.subplots(2, 1)
+        else:
+            fig, (ax1, ax2, ax3) = plt.subplots(3, 1)
+            ax3.set_title('Pair-wise effect')
+            ax3.bar(*zip(*self.pair_wise_effect['norm'].items()))
+
+        ax1.set_title('Linear effect')
+        ax1.bar(*zip(*self.linear_effect['norm'].items()))
+
+        ax2.set_title('Non-Linear effect')
+        ax2.bar(*zip(*self.non_linear_effect['norm'].items()))
+
+        fig.tight_layout()
+        return fig
+
+
+def _normalize(effect: dict) -> dict:
```

Comment:
This function is not included in the class, should we not move it there?

Reply:
Pylint recommends to make it static as it has no self. inside

Reply:
It can be a static method inside the class, use the decorator `@staticmethod`.